src/include/sof/audio/crossover/crossover.h

```diff
@@ -0,0 +1,125 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Google LLC. All rights reserved.
+ *
+ * Author: Sebastiano Carlucci <scarlucci@google.com>
+ */
+
+ #ifndef __SOF_AUDIO_CROSSOVER_CROSSOVER_H__
+ #define __SOF_AUDIO_CROSSOVER_CROSSOVER_H__
+
+#include <stdint.h>
+#include <sof/platform.h>
+#include <sof/audio/eq_iir/iir.h>
+#include <user/crossover.h>
+
+struct comp_buffer;
+struct comp_dev;
+
+#define CROSSOVER_MAX_STREAMS 4
+#define CROSSOVER_BIQUAD_COEF_SIZE (sizeof(int32_t) * 7)
+
+/**
+ * The Crossover filter will have from 2 to 4 outputs.
+ * Diagram of a 4-way Crossover filter (6 LR4 Filters).
+ *
+ *                             o---- LR4 LO-PASS --> y1(n)
+ *                             |
+ *          o--- LR4 LO-PASS --o
+ *          |                  |
+ *          |                  o--- LR4 HI-PASS --> y2(n)
+ * x(n) --- o
+ *          |                  o--- LR4 LO-PASS --> y3(n)
+ *          |                  |
+ *          o--- LR4 HI-PASS --o
+ *                             |
+ *                             o--- LR4 HI-PASS --> y4(n)
+ *
+ * Refer to include/user/crossover.h for diagrams of 2-way and 3-way crossovers
+ * The low and high pass LR4 filters have opposite phase responses, causing
+ * the intermediary outputs to be out of phase by 180 degrees.
+ * For 2-way and 3-way, the phases of the signals need to be synchronized.
+ *
+ * Each LR4 is made of two butterworth filters in series with the same params.
+ *
+ * x(n) --> BIQUAD --> z(n) --> BIQUAD --> y(n)
+ *
+ * In total, we keep track of the state of at most 6 IIRs each made of two
+ * biquads in series.
+ *
+ */
+
+struct lr4_state {
+	int32_t *coef;
```

Comment:
I think we need to move the individual biquad config/state into a struct so that we can just import that, that way if the ABI changes we will automatically inherit the change.